### DISCUSSION TITLE
WebComponents for custom display/interfaces

### DISCUSSION THREAD

**Discussion by telubeer** [directus/directus#9246] [Feature Requests]
Directus is a great tool! Big thanks for Directus team for the work you're doing.

I have an idea about the way to extend Admin UI more easily for some one who do not know Vue or to allow to use different tools for this task. And idea is to use WebComponents for rendering displays/interfaces. 

To get WebComponents work required:
1. load thirdparty javascript at Admin UI context some how (via direct link or installed npm package).
  1.1 additionally a place to define custom script for registering 
WebComponents if external script do not doing this for us.
  1.2 yes, it's a possible security breach. For lowering risks required to dump this script to local storage and validate against provided checksum. then we can attach validated script to global javascript context for usage.
2. A place where we can tell Vue to skip  registered WebComponents from processing `Vue.config.ignoredElements = [/custom-components-\w*/];`
3. A custom display/interface vue component to wrap a raw html layout where we can define markup for WebComponents usage
  3.1 wrapper should inject props with data/api instance  to defined web component and listen for possible events from it and transfer them to admin UI.

 I saw quick mention about web components at context of Discussion about support of React, but i do not want to use directus sdk each time to build a vue wrapper component when i need to use web component to create a custom display. 
A lot better, if i would just drop a link or npm package name and Directus will load WebComponent definisions for me. And this part of work should not be executed at custom display/interface level, because of security and performance reasons.

What do you think?

**Comment by rijkvanzanten** [comment]
> A lot better, if i would just drop a link or npm package name and Directus will load WebComponent definisions for me. And this part of work should not be executed at custom display/interface level, because of security and performance reasons.

I would agree with this. To me, it sounds like we'd want to make a standardized Vue wrapper component that you can simply attach to your custom web component to make it "directus compatible". I'm not totally sure what goes into that, but @nickrum might have some more ideas

**Reply by nickrum** [reply]
Having pre-built Vue wrapper components for each App extension type sounds like a nice way to implement this. This would be much cleaner and more flexible than adding "native" Web Component loading into Directus.
This would work very similar to the React wrapper someone from the community created. The tricky part would be to figure out how to inject things like the stores and the api into the Web Component.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
